sycl/test/CMakeLists.txt

```diff
@@ -15,7 +15,6 @@ set(SYCL_THREADS_LIB ${CMAKE_THREAD_LIBS_INIT})
 
 # TEST_INCLUDE_PATH is used for syntax-only verification of type information.
 list(APPEND test_includes ${SYCL_INCLUDE})
-list(APPEND test_includes ${SYCL_INCLUDE}/sycl)
```

Comment:
Note for reviewers: without this change `test/gdb/printer.cpp` failed to compile. The reason for that is because in `program_impl.hpp` we have `#include <detail/ur.hpp>` line.

We have two `detail/ur.hpp` headers: one under `sycl/include/sycl` and another under `sycl/source`. During the RT build we only have `sycl/include` and `sycl/source` include paths, so we take the right header and everything works fine.
However, for this specific test we add an extra include path, which makes a wrong file to be included and leads to missing declaration of `convertUrBackend` (which is moved into the library by this PR). I do not see any failures which would be caused by this particular change and I also don't quite understand why do we need this path at all.